nilearn/_utils/niimg_conversions.py

```diff
@@ -193,6 +207,28 @@ def check_niimg(niimg, ensure_ndim=None, atleast_4d=False, dtype=None,
     """
     from ..image import new_img_like  # avoid circular imports
 
+    if isinstance(niimg, _basestring):
+        if (wildcards and
+            ni.EXPAND_PATH_WILDCARDS):
+            filenames = sorted(glob.glob(niimg))  # Ascending sorting
+
+            # processing filenames matching globbing expression
+            if len(filenames) > 1:
+                niimg = filenames  # iterable case
+            elif len(filenames) == 1:
+                # Only one file matching => loading it as is
+                niimg = filenames[0]
```

Comment:
Maybe we should do: `elif [niimg] == filenames` to differentiate the case where niimg is just a filename vs the one where the glob only matches one file.  In the latter I think it should be considered as a list with a single file.